ncbi-amrfinderplus/3.12.8-2024-07-22.1/Dockerfile

```diff
@@ -0,0 +1,101 @@
+  tar zxf amrfinder_binaries_v${AMRFINDER_VER}.tar.gz && \
+  rm amrfinder_binaries_v${AMRFINDER_VER}.tar.gz && \
+  mkdir /data
+
+# install ncbi-blast linux binaries
+RUN echo "downloading ncbi-blast-${BLAST_VER}+ linux binaries from NCBI FTP..." && \
+  wget -q https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/${BLAST_VER}/ncbi-blast-${BLAST_VER}+-x64-linux.tar.gz && \
+  tar -xzf ncbi-blast-${BLAST_VER}+-x64-linux.tar.gz && \
+  rm -v ncbi-blast-${BLAST_VER}+-x64-linux.tar.gz
+
+# set PATH and locale settings for singularity compatibiliity, set amrfinder and manually-installed blast as higher priority in PATH
+ENV PATH="/amrfinder:/ncbi-blast-${BLAST_VER}+/bin:$PATH" \
+  LC_ALL=C
+
+# download databases and index them
+# done in this manner to pin the database version instead of pulling the latest version with `amrfinder -u` 
+# softlink is required for `amrfinder -l` and typical `amrfinder` use cases to work properly
+RUN mkdir -p /amrfinder/data/${AMRFINDER_DB_VER} && \
+  wget -q -P /amrfinder/data/${AMRFINDER_DB_VER} ftp://ftp.ncbi.nlm.nih.gov/pathogen/Antimicrobial_resistance/AMRFinderPlus/database/3.12/${AMRFINDER_DB_VER}/* && \
+  amrfinder_index /amrfinder/data/${AMRFINDER_DB_VER} && \
+  ln -s /amrfinder/data/${AMRFINDER_DB_VER} /amrfinder/data/latest
+
+# set final working directory
+WORKDIR /data
+
+# default command is to print help options
+CMD [ "amrfinder", "--help" ]
+
+## Test stage
+FROM app as test
+
+# list database version and available --organism options
+RUN amrfinder -l
+
+# run recommended tests from amrfinder
+RUN amrfinder --threads 1 --plus -p /amrfinder/test_prot.fa -g  /amrfinder/test_prot.gff -O Escherichia --print_node > test_prot.got && \
+  diff /amrfinder/test_prot.expected test_prot.got && \
+  amrfinder --threads 1 --plus -n /amrfinder/test_dna.fa -O Escherichia --print_node > test_dna.got && \
+  diff /amrfinder/test_dna.expected test_dna.got && \
+  amrfinder --threads 1 --plus -n /amrfinder/test_dna.fa -p /amrfinder/test_prot.fa -g /amrfinder/test_prot.gff -O Escherichia --print_node > test_both.got && \
+  diff /amrfinder/test_both.expected test_both.got
+
+# run amrfinder on Salmonella, without and with --organism option
+RUN wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/010/941/835/GCA_010941835.1_PDT000052640.3/GCA_010941835.1_PDT000052640.3_genomic.fna.gz  && \
+  gzip -d GCA_010941835.1_PDT000052640.3_genomic.fna.gz && \
+  amrfinder --threads 4 --plus --nucleotide GCA_010941835.1_PDT000052640.3_genomic.fna --output test1.txt && \
+  amrfinder --threads 4 --plus --nucleotide GCA_010941835.1_PDT000052640.3_genomic.fna --organism Salmonella --output test2.txt && \
+  cat test1.txt test2.txt
+
+# run amrfinder on Klebesiella oxytoca using --organism/-O flag
+RUN wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/003/812/925/GCA_003812925.1_ASM381292v1/GCA_003812925.1_ASM381292v1_genomic.fna.gz && \
+  gzip -d GCA_003812925.1_ASM381292v1_genomic.fna.gz && \
```

Comment:
I recently learned that amrfinderplus can accept gzipped assemblies. Could you remove lines 97 and 90 & update the below lines to use the assembly that is gzipped?

Reply:
I didn't know that. Nice feature, thank you!